File: cpp/Insert_Interval.cpp
/**
 * Definition for an interval.
 * struct Interval {
 *     int start;
 *     int end;
 *     Interval() : start(0), end(0) {}
 *     Interval(int s, int e) : start(s), end(e) {}
 * };
 */
class Solution {
public:
    vector<Interval> insert(vector<Interval> &intervals, Interval newInterval) {
        // Note: The Solution object is instantiated only once and is reused by each test case.
        Interval newInt = newInterval;
        vector<Interval> ret;
        bool inserted = false;
        for (int n = 0; n < intervals.size(); n ++){
            Interval cur = intervals[n];
            if (cur.start > newInt.end) {
                if (inserted == false) {
                    ret.push_back(newInt);
                    inserted = true;
                }
                ret.push_back(cur);
            }
            else if (cur.end < newInt.start) {
                ret.push_back(cur);
            }
            else {
                newInt.start = min(cur.start, newInt.start);
                newInt.end = max(cur.end, newInt.end);
            }
        }
        if (inserted == false) {
            ret.push_back(newInt);
            inserted = true;
        }
        return ret;
    }
};

File: Python/Insert_Interval.py
# Definition for an interval.
# class Interval:
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution:
    # @param intervals, a list of Intervals
    # @param newInterval, a Interval
    # @return a list of Interval
    def insert(self, intervals, newInterval):
        ret = []
        inserted = False
        for itv in intervals:
            if itv.end < newInterval.start:
                ret.append(itv)
            elif itv.start > newInterval.end:
                if not inserted:
                    ret.append(newInterval)
                    inserted = True
                ret.append(itv)
            else:
                newInterval.start = min(itv.start, newInterval.start)
                newInterval.end = max(itv.end, newInterval.end)
        if not inserted: ret.append(newInterval)
        return ret
